### DISCUSSION TITLE
How to see all possible typehints in Kernel.GetInputAsync

### DISCUSSION THREAD

**Discussion by zz415** [dotnet/interactive#2704] [Q&A]
Is there a way to see the full list of eligable type hints when using Kernel.GetInputAsync()?  The examples in the input prompt documentation (https://github.com/dotnet/interactive/blob/main/docs/input-prompts.md) only show for string where you don't specify a typehint or typehint = "file".  What is the full list of valid typehints besides "file",  is there one to specify as folder/directory as opposed to file?  Is there one to specify file type ? "file:xlsx" ?  appreciate the help on this, I have been looking at all the source code and can't figure it out.

**Comment by jonsequitur** [comment]
We could improve the documentation but I don't know that there's a better approach. The answer will be different from one frontend to another. Right now in VS Code we only support `text` (the default and fallback) and `file`. Only `text` is supported in other frontends, such as JupyterLab, where we're not currently investing in building extensions. (We do plan to add more in VS Code though.) 

This inconsistency across frontends is by design. Type hints are intended to be interpreted by the frontend and are based on the types supported by the HTML [`input` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types), which result in slightly different experiences and different levels of support among browsers.

**Reply by zz415** [reply]
hi jonsequitur,

thanks for the quick reply, I currently use VS Code,

I have built a lot of similar stuff out of python w/jupyter notebooks that utilizes the tkinter library to specify one file, multiple files, folder location/directory , and calendar date picking.

I'm a big fan of vscode notebook, only thing I would probably add is the ability to specify 1 vs multiple files or choosing a location/directory.

thanks again
